### PR TITLE
[EG] Beta One Client

### PR DESCRIPTION
**Pull request by l0lawrence**
EventGrid Beta Feature Work

One client approach - removal of EventGridPublisher mentions in docs/samples/readme

EventGridClient can now be used for both EventGrid Basic and EventGrid Namespaces

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_patch.py [hunk 1]
```diff
@@ -6,13 +6,94 @@
...
+DEFAULT_STANDARD_API_VERSION = "2023-10-01-preview"
+DEFAULT_BASIC_API_VERSION = "2018-01-01"
+
+
+class EventGridClient(InternalEventGridClient):
+    """Azure Messaging EventGrid Client.
+    :param endpoint: The host name of the namespace, e.g.
+     namespaceName1.westus-1.eventgrid.azure.net. Required.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure. Is either a
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_patch.py:45]: probably don't need to say the type in the description if it's under the type docstring?

File: sdk/eventgrid/azure-eventgrid/README.md [hunk 2]
```diff
@@ -86,68 +155,43 @@ pass the key as a string into an instance of [AzureKeyCredential][azure-key-cred
...
-
-Event Grid supports multiple schemas for encoding events.
-
-#### Event Grid schema
-While you may configure your topic to use a [custom schema](https://docs.microsoft.com/azure/event-grid/input-mappings), it is more common to use the already-defined Event Grid schema. See the specifications and requirements [here](https://docs.microsoft.com/azure/event-grid/event-schema).
-
-#### CloudEvents v1.0 schema
-Another option is to use the CloudEvents v1.0 schema. [CloudEvents](https://cloudevents.io/) is a Cloud Native Computing Foundation project which produces a specification for describing event data in a common way. The service summary of CloudEvents can be found [here](https://docs.microsoft.com/azure/event-grid/cloud-event-schema).
+If you are using Event Grid Basic, regardless of the schema that your topic or domain is configured to use, `EventGridClient` will be used to publish events to it. Use the `send` method to publish events.
```
[the remark this answers is not in the document]
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/README.md:178]: this already existed in the previous readme for basic

File: sdk/eventgrid/azure-eventgrid/README.md [hunk 3]
```diff
@@ -197,7 +241,7 @@ event = EventGridEvent(
 )
 
 credential = AzureKeyCredential(key)
-client = EventGridPublisherClient(endpoint, credential)
+client = EventGridClient(endpoint, credential, level=ClientLevel.BASIC)
```
[the remark this answers is not in the document]
Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/README.md:244]: all these examples are only supported by EventGrid Basic

**Comment by l0lawrence** [comment]
/azp python run - eventgrid - tests

File: sdk/eventgrid/azure-eventgrid/CHANGELOG.md [hunk 4]
```diff
@@ -4,11 +4,15 @@
...
 ### Features Added
 
-### Breaking Changes
+- This is a Beta of the EventGridClient
+  - EventGridClient `send` can be used for both Event Grid Namespace Resources and Event Grid Basic Resources.
+      - Added a kwarg `level` in the EventGridClient constructor. The default value is `Standard` which creates a client for an Event Grid Namespace Resource.
+
+ ### Breaking Changes 
+  - Removed the `AcknowledgeOptions`,`ReleaseOptions`, `RejectOptions`, and `RenewLockOptions` models. `lock_tokens` can now be specified as a `kwarg` on the operation.
+  - Renamed `publish_cloud_events` to `send`.
```
Comment (l0lawrence) [inline sdk/eventgrid/azure-eventgrid/CHANGELOG.md:13]: will add other renames here as well :)

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_patch.py [hunk 5]
```diff
@@ -6,8 +6,83 @@
...
+    :param endpoint: The endpoint to the Event Grid resource.
+    :type endpoint: str
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.AzureSasCredential or
+     ~azure.core.credentials_async.AsyncTokenCredential
+    :keyword api_version: The API version to use for this operation. Default value for namespaces is
+     "2023-10-01-preview". Default value for basic is "2018-01-01". Note that overriding this default value may result in unsupported
+     behavior.
+    :paramtype api_version: str or None
+    :keyword level: The level of the event grid service.
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_patch.py:39]: nit: for consistency
```suggestion
    :keyword level: The level of the Event Grid service.
```

**Review by swathipil — APPROVED** [review]
lgtm!

**Pull request merged**